lib/vagrant-service-manager/command.rb

```diff
@@ -101,36 +110,55 @@ def print_help(config = {})
         exit config[:exit_status]
       end
 
-      def check_if_a_service_is_running?(service)
+      def service_running?(service)
         command = "systemctl status #{service}"
         with_target_vms(nil, {:single_target=>true}) do |machine|
           return machine.communicate.test(command)
         end
       end
 
-      def print_all_provider_info
-        @env.ui.info I18n.t('servicemanager.commands.env.nil')
+      def execute_status_display(service = nil)
+        if service
+          status = if service_running?(service)
+                     I18n.t('servicemanager.commands.status.status.running')
+                   else
+                     I18n.t('servicemanager.commands.status.status.stopped')
+                   end
+          @env.ui.info("#{service} - #{status}")
+        else
+          @env.ui.info I18n.t('servicemanager.commands.status.nil')
+          running_services = []
+          SUPPORTED_SERVICES.each do |service|
+            status = if service_running?(service)
+                       running_services << service
+                       I18n.t('servicemanager.commands.status.status.running')
+                      else
+                       I18n.t('servicemanager.commands.status.status.stopped')
+                      end
+            @env.ui.info("#{service} - #{status}")
+          end
+        end
+      end
 
+      def get_running_services
         running_services = []
         SUPPORTED_SERVICES.each do |service|
-          status = if check_if_a_service_is_running?(service)
-                     running_services << service
-                     I18n.t('servicemanager.commands.env.status.running')
-                    else
-                     I18n.t('servicemanager.commands.env.status.stopped')
-                    end
-          @env.ui.info("#{service} - #{status}")
+          if service_running?(service)
```

Comment:
`running_services << service if service_running?(service)` is what you may prefer if there is only one line `if` block